src/Directions.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react'
+
+import { MapContext } from './Map'
+
+import { createCoordinate, NumberTuple } from './utils'
+
+type annotationOptions = {
+  origin?: mapkit.MarkerAnnotationConstructorOptions
+  destination?: mapkit.MarkerAnnotationConstructorOptions
+}
+
+type DirectionsProps = {
+  origin: NumberTuple
+  destination: NumberTuple
+  transportType?: 'Automobile' | 'Walking'
+  style?: mapkit.StyleConstructorOptions
+  options?: mapkit.DirectionsConstructorOptions
+  showAnnotations?: boolean
+  annotationOptions?: annotationOptions
+  routeDetails?: (arg0: {
+    steps: mapkit.RouteStep[]
+    distance: number
+    expectedTravelTime: number
+  }) => void
+}
+
+export const Directions: React.FC<DirectionsProps> = ({
+  origin,
+  destination,
+  transportType = 'Automobile',
+  style,
+  showAnnotations = true,
+  annotationOptions = {
+    origin: { color: 'red' },
+    destination: { color: 'green' },
+  },
+  options,
+  routeDetails = () => null,
+}) => {
+  const { mapkit, map } = React.useContext(MapContext)
+
+  React.useEffect(() => {
+    if (mapkit && map) {
+      const directions = new mapkit.Directions(options)
+      const originCoordinates = createCoordinate(
+        origin[0] as number,
```

Comment:
are these annotations required for some reason? I think this can be inferred from NumberTuple

Reply:
Is the suggestion here to remove the type casting to number?

```suggestion
        origin[0],
```